runner/runners/queue.go

```diff
@@ -0,0 +1,116 @@
+package runners
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/scootdev/scoot/runner"
+	"github.com/scootdev/scoot/runner/execer"
+	"github.com/scootdev/scoot/snapshot"
+)
+
+const QueueFullMsg = "No resources available. Please try later."
+
+// commandAndID is a command waiting to run in the queue and the ID we've assigned
+type commandAndID struct {
+	cmd *runner.Command
+	id  runner.RunId
+}
+
+// NewQueueRunner creates a new runner with a queue of capacity
+func NewQueueRunner(exec execer.Execer, filer snapshot.Filer, outputCreator runner.OutputCreator, capacity int) runner.Runner {
+	statuses := NewStatuses()
+	invoker := NewInvoker(exec, filer, outputCreator)
+	controller := &QueueController{statuses: statuses, invoker: invoker, capacity: capacity}
+	return NewControllerAndStatuserRunner(controller, statuses)
+}
+
+// TODO(dbentley): QueueController and SingleController are so similar...
+
+// QueueController allows only one-at-a-time, but also allows queuing up to capacity cmds.
+type QueueController struct {
+	statuses *Statuses
+	invoker  *Invoker
+	capacity int
+
+	runningID runner.RunId
+	abortCh   chan struct{}
+	queue     []commandAndID
+	mu        sync.Mutex
+}
+
```

Comment:
It seems odd to have status control the runId - I understand that it makes sense when thinking about runId as a db key, but it just makes me pause.